### DISCUSSION TITLE
Create custom spinner

### DISCUSSION THREAD

**Discussion by redigaffi** [quasarframework/quasar#12773] [General - Components / Directives / etc]
Hello, 

Is it possible to create a custom spinner? If yes, how can that be done?

**Comment by AdagioB** [comment]
Unfortunately it doesn't look like it's possible. I was also looking for a solution, but so far all my search has come up empty

**Comment by yusufkandemir** [comment]
The simplest spinner implementation in terms of SVG structure is `QSpinner`: https://github.com/quasarframework/quasar/blob/30e4cd81502b753e64f07b69a5731818189d8fac/ui/src/components/spinner/QSpinner.js

Underlying composable(`useSpinner`): https://github.com/quasarframework/quasar/blob/30e4cd81502b753e64f07b69a5731818189d8fac/ui/src/components/spinner/use-spinner.js

The composable is not exposed, but you don't really need it. It is made composable in Quasar's source code because there are tons of different spinners([_e.g. `QSpinnerBall`, `QSpinnerGears`, etc._](https://github.com/quasarframework/quasar/tree/30e4cd81502b753e64f07b69a5731818189d8fac/ui/src/components/spinner)).

So, we can write a custom spinner identical to the `QSpinner` I've sent above:
```js
import { computed, defineComponent, h } from 'vue'

const sizeDefaults = {
  xs: 18,
  sm: 24,
  md: 32,
  lg: 38,
  xl: 46
}

export default defineComponent({
  name: 'CustomSpinner',

  props: {
    size: {
      type: [ Number, String ],
      default: '1em'
    },
    color: String,

    thickness: {
      type: Number,
      default: 5
    }
  },

  setup (props) {
    const cSize = computed(() => (
      props.size in sizeDefaults
        ? `${ sizeDefaults[ props.size ] }px`
        : props.size
    ))

    const classes = computed(() =>
      'q-spinner' + (props.color ? ` text-${ props.color }` : '')
    )

    return () => h('svg', {
      class: classes.value + ' q-spinner-mat',
      width: cSize.value,
      height: cSize.value,
      viewBox: '25 25 50 50'
    }, [
      h('circle', {
        class: 'path',
        cx: '50',
        cy: '50',
        r: '20',
        fill: 'none',
        stroke: 'currentColor',
        'stroke-width': props.thickness,
        'stroke-miterlimit': '10'
      })
    ])
  }
})
```

You can check out other spinners for different SVG structures: https://github.com/quasarframework/quasar/tree/30e4cd81502b753e64f07b69a5731818189d8fac/ui/src/components/spinner

**Comment by suleimanas** [comment]
Yes you can, it's just an svg inside a dev, you can make your own spinner component or customize quasar's spinners css.
also you can use external spinners like spinkit
